dbt/adapters/dremio/api/rest/entities/reflection.py

```diff
@@ -0,0 +1,141 @@
+from enum import Enum
+
+
+class TransformType(Enum):
+    YEAR = "YEAR"
+    MONTH = "MONTH"
+    DAY = "DAY"
+    HOUR = "HOUR"
+    IDENTITY = "IDENTITY"
+    BUCKET = "BUCKET"
+    TRUNCATE = "TRUNCATE"
+
+    @classmethod
+    def from_string(cls, transform_str):
+        transform_str = transform_str.upper()
+
+        if transform_str.startswith("BUCKET("):
+            return cls.BUCKET
+        elif transform_str.startswith("TRUNCATE("):
+            return cls.TRUNCATE
+
+        try:
+            return cls(transform_str)
+        except ValueError:
+            return cls.IDENTITY
+
+    def to_transform(self, raw_str):
+        if self in (
+                TransformType.YEAR,
+                TransformType.MONTH,
+                TransformType.DAY,
+                TransformType.HOUR,
+                TransformType.IDENTITY
+        ):
+            return {"type": self.value}
+
+        if self == TransformType.BUCKET:
+            bucket_count = int(raw_str.split("(")[1].split(")")[0])
+            return {
+                "type": "BUCKET",
+                "bucketTransform": {"bucketCount": bucket_count},
+            }
+
+        if self == TransformType.TRUNCATE:
+            truncate_length = int(raw_str.split("(")[1].split(")")[0])
+            return {
+                "type": "TRUNCATE",
+                "truncateTransform": {"truncateLength": truncate_length},
+            }
+
+        return {"type": TransformType.IDENTITY.value}
+
+
+# https://docs.dremio.com/24.3.x/reference/api/reflections/
```

Comment:
Should we have latest version of the docs link? Without 24.3.x version?

Reply:
Let's use standard python documentation way next time. https://realpython.com/documenting-python-code/
It's good to start describing what do we have in the implementation and why do we have it.